packages/providers/src/fuel-graphql-request.ts

```diff
@@ -0,0 +1,154 @@
+/* 
+
+  Replaces `graphql-request` and `graphql-tag` dependencies with a simple
+  `fetch` implementation, reducing the final bundle size by 15.68 kB.
+
+  More:
+    https://github.com/FuelLabs/fuels-ts/discussions/1592#discussioncomment-8003515
+
+*/
+
+import { FuelError } from '@fuel-ts/errors';
+import type { DocumentNode } from 'graphql';
+import { print } from 'graphql';
+
+function isSubscription({ definitions: defs }: DocumentNode) {
+  const opDef = defs.find((op) => op.kind === 'OperationDefinition');
+
+  if (opDef && 'operation' in opDef) {
+    return opDef?.operation === 'subscription';
+  }
+
+  return false;
+}
+
+class FuelGraphqlSubscriber implements AsyncIterator<unknown> {
+  private stream!: ReadableStreamDefaultReader<Uint8Array>;
+  private static textDecoder = new TextDecoder();
+
+  private static parseBytesStream(
+    bytes: Uint8Array | undefined
+  ): Record<string, unknown> | FuelError | undefined {
+    if (bytes === undefined) {
+      return undefined;
+    }
+
+    const text = this.textDecoder.decode(bytes);
+
+    // https://github.com/FuelLabs/fuel-core/blob/e1e631902f762081d2124d9c457ddfe13ac366dc/crates/fuel-core/src/graphql_api/service.rs#L247
+    if (text === 'keep-alive-text') {
+      return undefined;
+    }
+
+    const { data, errors } = JSON.parse(text.split('data:')[1]);
```

Comment:
Is it guaranteed that the value of `text.split('data:')[1]` will always be parseable? 

Can this value be `undefined` or any other value that will make `JSON.parse` throws?

If so, we must add a `Try/Catch` block here.